components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/ai/BrandingAIPreferenceManagerImpl.java

```diff
@@ -0,0 +1,293 @@
+    private static final String BRANDING_AI_STATUS_ENDPOINT = "/api/server/v1/branding-preference/status";
+    private static final String BRANDING_AI_RESULT_ENDPOINT = "/api/server/v1/branding-preference/result";
+
+    private static final Log LOG = LogFactory.getLog(BrandingAIPreferenceManagerImpl.class);
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    /**
+     * This method will connect to the BrandingAI microservice and generate the branding preference for the given
+     * website URL.
+     *
+     * @param websiteURL Website URL for which the branding preference should be generated.
+     * @return Operation ID of the branding preference generation.
+     * @throws BrandingAIServerException When errors occurs from the AI service.
+     * @throws BrandingAIClientException When invalid request is sent to the AI service.
+     */
+    @Override
+    public String generateBrandingPreference(String websiteURL) throws BrandingAIServerException,
+            BrandingAIClientException {
+
+        Map<String, String> requestBody = new HashMap<>();
+        requestBody.put("website_url", websiteURL);
+        Object response = executeRequest(BRANDING_AI_GENERATE_ENDPOINT, HttpPost.class, requestBody);
+        return ((Map<String, String>) response).get("operation_id");
+    }
+
+    /**
+     * This method will connect to the BrandingAI microservice and get the status of the branding preference generation.
+     *
+     * @param operationId Operation ID of the branding preference generation.
+     * @return Status of the branding preference generation.
+     * @throws BrandingAIServerException When errors occurs from the AI service.
+     * @throws BrandingAIClientException When invalid request is sent to the AI service.
+     */
+    @Override
+    public Object getBrandingPreferenceGenerationStatus(String operationId) throws BrandingAIServerException,
+            BrandingAIClientException {
+
+        return executeRequest(BRANDING_AI_STATUS_ENDPOINT + "/" + operationId, HttpGet.class, null);
+    }
+
+    /**
+     * This method will connect to the BrandingAI microservice and get the result of the branding preference generation.
+     *
+     * @param operationId Operation ID of the branding preference generation.
+     * @return Result of the branding preference generation.
+     * @throws BrandingAIServerException When errors occurs from the AI service.
+     * @throws BrandingAIClientException When invalid request is sent to the AI service.
+     */
+    @Override
+    public Object getBrandingPreferenceGenerationResult(String operationId) throws BrandingAIServerException,
+            BrandingAIClientException {
+
+        return executeRequest(BRANDING_AI_RESULT_ENDPOINT + "/" + operationId, HttpGet.class, null);
+    }
+
+
+    private Object executeRequest(String endpoint, Class<? extends HttpUriRequest> requestType, Object requestBody)
```

Comment:
Let's add a method comment here as well

Reply:
Are we adding methods comments to private methods?